cirq-core/cirq/circuits/circuit.py

```diff
@@ -1820,7 +1820,11 @@ def _load_contents_with_earliest_strategy(self, contents: 'cirq.OP_TREE'):
         # each index.
         for i in range(length):
             if i in moments_by_index:
-                self._moments.append(moments_by_index[i].with_operations(op_lists_by_index[i]))
+                if op_lists_by_index[i]:
+                    self._moments.append(moments_by_index[i].with_operations(op_lists_by_index[i]))
```

Comment:
Perhaps we could modify `Moment.with_operations` instead, so that we can get this optimization in more scenarios, e.g.
```python
    def with_operations(self, *contents: 'cirq.OP_TREE') -> 'cirq.Moment':
        ...
        flattened_contents = tuple(op_tree.flatten_to_ops(contents))
        if not flattened_contents:
            return self
        ...
```

Reply:
Will revert this in the next commit,